Update README.md (#17978)
MetaMask/metamask-extension: README.md

- Install [Node.js](https://nodejs.org) version 16
  - If you are using [nvm](https://github.com/nvm-sh/nvm#installing-and-updating) (recommended) running `nvm use` will automatically choose the right node version for you.
- Install [Yarn v3](https://yarnpkg.com/getting-started/install)

>     - Note: follow the steps for installing and updating the `yarn` version, but NOT the step for "Initializing your project". You should not run `yarn init -2`, because that wipe the metamask repo.  

Reviewer: The later sections of this page will also be misleading. e.g. `yarn set version stable` will change the Yarn version used.

Maybe instead we should advise continuing only up to the "Updating the global Yarn version" section, and going no further

Author: Good suggestion. Updated in the latest commit

- Install [Node.js](https://nodejs.org) version 16
  - If you are using [nvm](https://github.com/nvm-sh/nvm#installing-and-updating) (recommended) running `nvm use` will automatically choose the right node version for you.
- Install [Yarn v3](https://yarnpkg.com/getting-started/install)
    - ONLY follow the steps in the "Install Corepack" and "Updating the global Yarn version" sections

>     - DO NOT take any of the steps in the "Initializing your project", "Updating to the latest versions" or "Installing the latest build fresh from master" sections. These steps could result in your repo being reset or installing the wrong yarn version, which can break your build.

Reviewer: This might be easier to understand:
```suggestion
    - DO NOT take any of the steps in the "Initializing your project" section or later sections. These steps could result in your repo being reset or installing the wrong yarn version, which can break your build.
```
